pkg/machines/components/vm/vmExpandedContent.jsx

```diff
@@ -27,40 +27,65 @@ import VmNetworkTab from '../vmnetworktab.jsx';
-        { name: consolesTabName, id: cockpit.format("$0-consoles", vmId(vm.name)), renderer: Consoles, data: { vm, config, dispatch, onAddErrorNotification } },
-    ];
+        this.state = {
+            snapshotsAPIsAvailable: false,
+        };
+    }
 
-    let initiallyActiveTab = null;
-    if (vm.ui && vm.ui.initiallyOpenedConsoleTab) {
-        initiallyActiveTab = tabRenderers.map((o) => o.name).indexOf(consolesTabName);
+    componentDidMount() {
+        cockpit.spawn(["gdbus", "introspect", "--session", "--dest", "org.libvirt", "--object-path", "/org/libvirt/QEMU"])
+                .catch(e => console.log(e))
+                .done(output => {
+                    if (output.includes("snapshot"))
+                        this.setState({ snapshotsAPIsAvailable: true });
+                });
     }
 
-    return (<ListingPanel
-        colSpan='4'
-        initiallyActiveTab={initiallyActiveTab}
-        tabRenderers={tabRenderers} />);
-};
+    render() {
+        const {
+            vm, vms, config, libvirtVersion, hostDevices, storagePools,
+            onUsageStartPolling, onUsageStopPolling, dispatch, networks,
+            interfaces, nodeDevices, onAddErrorNotification
+        } = this.props;
+
+        const overviewTabName = (<a href="#" id={`${vmId(vm.name)}-overview`}>{_("Overview")}</a>);
+        const usageTabName = (<a href="#" id={`${vmId(vm.name)}-usage`}>{_("Usage")}</a>);
+        const disksTabName = (<a href="#" id={`${vmId(vm.name)}-disks`}>{_("Disks")}</a>);
+        const networkTabName = (<a href="#" id={`${vmId(vm.name)}-networks`}>{_("Network Interfaces")}</a>);
+        const snapshotsTabName = (<a href="#" id={`${vmId(vm.name)}-snapshots`}>{_("Snapshots")}</a>);
+        const consolesTabName = (<a href="#" id={`${vmId(vm.name)}-consoles`}>{_("Consoles")}</a>);
+
+        const tabRenderers = [
+            { name: overviewTabName, id: cockpit.format("$0-overview", vmId(vm.name)), renderer: VmOverviewTab, data: { vm, config, dispatch, nodeDevices, libvirtVersion } },
+            { name: usageTabName, id: cockpit.format("$0-usage", vmId(vm.name)), renderer: VmUsageTab, data: { vm, onUsageStartPolling, onUsageStopPolling }, presence: 'onlyActive' },
+            { name: disksTabName, id: cockpit.format("$0-disks", vmId(vm.name)), renderer: VmDisksTab, data: { vm, vms, config, storagePools, onUsageStartPolling, onUsageStopPolling, dispatch, onAddErrorNotification }, presence: 'onlyActive' },
+            { name: networkTabName, id: cockpit.format("$0-networks", vmId(vm.name)), renderer: VmNetworkTab, presence: 'onlyActive', data: { vm, dispatch, config, hostDevices, interfaces, networks, nodeDevices, onAddErrorNotification } },
+            { name: snapshotsTabName, id: cockpit.format("$0-snapshots", vmId(vm.name)), renderer: VmSnapshotsTab, hide: !this.state.snapshotsAPIsAvailable, data: { vm, dispatch, config, onAddErrorNotification } },
```

Comment:
Please just don't add this entry at all if the API is not available, then you don't need the (rather un-React-y) `hide` property.

Reply:
I explained problem in another comment concerning activeTab index. I guess I have to do much more code rewriting in the ListingPanel to avoid that problem.

Reply:
Ack, let's keep that approach then.